rhel/containers/matcher.go

```diff
@@ -0,0 +1,61 @@
+package containers
+
+import (
+	"context"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"golang.org/x/mod/semver"
```

Comment:
Do Red Hat containers actually use semver and not RPM versions?

Reply:
Well, each tag begins with a "v", and all tags I've looked at comply with Semantic Version rules. However I haven't actually read anywhere that Red Hat containers, or container registry tags in general do use strict Semantic Versioning.

I'm fleshing this out further in a following commit, which creates vulnerabilities with affects ranges for each minor version. I'm creating a custom type for handing them, but using x/mod/semver still for parsing the tags.

Reply:
I just noticed that the UBI container doesn't use semantic versioning. Luckily that container doesn't have any container-first content, as it's comprised of RPMs only.

I'm going to stick with rhctag.Version for now, and just skip containers tag which we can't parse as Semantic Versions. We can revisit this if it becomes a problem.

Reply:
BTW, I removed x/mod/semver import from this package. It's only in the pkg/rhctag package now.